### PR TITLE
Prevents the map reader from resetting when another map is read

### PR DESCRIPTION
**Pull request by Crazylemon64**
:cl: Crazylemon
fix: The map loader no longer resets when it reads another map mid-load
/:cl:

### CODE REVIEW

File: code/modules/awaymissions/maploader/reader.dm [hunk 1]
```diff
@@ -41,8 +41,8 @@ var/global/dmm_suite/preloader/_preloader = new
 	var/list/grid_models = list()
 	var/key_len = 0
 
-	dmmRegex.next = 1
-	while(dmmRegex.Find(tfile, dmmRegex.next))
+	var/regex_index = 1
+	while(dmmRegex.Find(tfile, regex_index))
```
Comment (CrAzYPiLoT-SS13) [inline code/modules/awaymissions/maploader/reader.dm]: I know my understanding of the map reader is very limited, and I am probably wrong, but, do you ever increment the value of `regex_index`?

Comment (monster860) [inline code/modules/awaymissions/maploader/reader.dm]: Yeah, shouldn't you be setting the value of `regex_index` to `dmmRegex.next` first thing in the while loop

Comment (Crazylemon64) [inline code/modules/awaymissions/maploader/reader.dm]: Oh, yes, I forgot that line. That line is VERY VERY important.

Comment (CrAzYPiLoT-SS13) [inline code/modules/awaymissions/maploader/reader.dm]: Good thing we have such professional maintainers :^)

**Comment by CrAzYPiLoT-SS13** [comment]
I'd say this is good for merging

**Pull request merged**